include/nuttx/mm/vm_map.h

```diff
@@ -0,0 +1,176 @@
+/****************************************************************************
+ * include/nuttx/mm/vm_map.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __MM_VM_MAP_VM_MAP_H
+#define __MM_VM_MAP_VM_MAP_H
+
+#ifdef CONFIG_MM_VM_MAP
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/fs/fs.h>
+#include <queue.h>
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+/* Type of the mapping */
+
+enum vm_map_type
+{
+  VM_MAP_ANONYMOUS,
+  VM_MAP_NAMED,
+  VM_MAP_FILE,
+  VM_MAP_SHM,
+  VM_MAP_RAMMAP,
+};
+
+/* Id of the mapping */
+
+union vm_map_id
+{
+  FAR const char *name;
+  FAR struct inode *inode;
+  int shmid;
+  uintptr_t val;
+};
+
+/* A vm mapping list item */
+
+struct vm_map_entry
```

Comment:
as discussed before, rammaps are note replacable by this. I changed the sm to use vm_map mappings already. The whole group_shm_s is not removed, since it also contains an allocator handle in addition to the mapping info. I didn't plan to add support for storing process-specific allocators in the same vm_map structure, but it might make sense conceptually. If you insist, I can do that; although this would then again widen the scope of this PR.

Reply:
> as discussed before, rammaps are note replacable by this.

Sorry, rammaps could be replaced.

> I changed the sm to use vm_map mappings already. The whole group_shm_s is not removed, since it also contains an allocator handle in addition to the mapping info. I didn't plan to add support for storing process-specific allocators in the same vm_map structure, but it might make sense conceptually. If you insist, I can do that; although this would then again widen the scope of this PR.

The change look reasonable, but how about we remove group_shm_s and move gs_handle into group_s directly?

Reply:
I can do that, I initially added an interface for adding memory allocators into the same vm_map, I am pushing this one first; if you think that it is better to keep the gs_handle separately in group_s, I will change.